collectors/python.d.plugin/pandas/pandas.chart.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Description: pandas netdata python.d module
+# Author: Andrew Maguire (andrewm4894)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import pandas as pd
+import requests
+
+from bases.FrameworkServices.SimpleService import SimpleService
+
+ORDER = []
+
+CHARTS = {}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.chart_configs = self.configuration.get('chart_configs', None)
+        self.line_sep = self.configuration.get('line_sep', ';')
+
+    def run_code(self, df_steps):
+        """eval() each line of code and ensure the result is a pandas dataframe"""
+
+        # process each line of code
+        lines = df_steps.split(self.line_sep)
+        for line in lines:
+            line_clean = line.strip('\n').strip(' ')
+            if line_clean != '' and line_clean[0] != '#':
+                df = eval(line_clean)
+                assert isinstance(df, pd.DataFrame), 'The result of each evaluated line of `df_steps` must be of type `pd.DataFrame`'
+
+        # take top row of final df as data to be collected by netdata
+        data = df.to_dict(orient='records')[0]
+
+        return data
+
+    def check(self):
+        """ensure charts and dims all configured and that we can get data"""
+
+        data = dict()
+
+        # add each chart as defined by the config
+        for chart_config in self.chart_configs:
```

Comment:
Exception by default (self.chart_configs is None)

human-readable message better than `TypeError: 'NoneType' object is not iterable`

Reply:
added a check into `check()`